core/server/services/auth/api-key/admin.js

```diff
@@ -0,0 +1,100 @@
+const jwt = require('jsonwebtoken');
+const models = require('../../../models');
+const common = require('../../../lib/common');
+
+const JWT_OPTIONS = {
+    maxAge: '5m',
+    algorithms: ['HS256']
+};
+
+/**
+ * Remove 'Bearer' from raw authorization header and extract the JWT token.
+ * Eg. Authorization: Bearer ${JWT}
+ * @param {string} header
+ */
+const extractTokenFromHeader = function extractTokenFromHeader(header) {
+    const [scheme, token] = header.split(' ');
+
+    if (/^Bearer$/i.test(scheme)) {
+        return token;
+    }
+
+    return;
+};
+
+const authenticateAdminApiKey = function authenticateAdminApiKey(req, res, next) {
+    if (req.query && req.query.content_key) {
```

Comment:
Should this be a separate piece of middleware? Conceptually we should be able to add this middleware to any route, without affecting other auth mechanisms :thinking:

Reply:
This fn is in general very long and hard to follow what it does without having a function comment which describes the steps.